desc/compute/_profiles.py

```diff
@@ -644,40 +615,212 @@ def _iota_rr(params, transforms, profiles, data, **kwargs):
             - 2 * current_term_r * data["psi_rr"]
             - current_term * data["psi_rrr"]
         ) / data["psi_r"]
-        num = (
-            data["lambda_z"] * data["g_tt"] - (1 + data["lambda_t"]) * data["g_tz"]
-        ) / data["sqrt(g)"]
-        num_r = (
-            data["lambda_rz"] * data["g_tt"]
-            + data["lambda_z"] * data["g_tt_r"]
-            - data["lambda_rt"] * data["g_tz"]
-            - (1 + data["lambda_t"]) * data["g_tz_r"]
-        ) / data["sqrt(g)"] - num * data["sqrt(g)_r"] / data["sqrt(g)"]
-        num_rr = (
-            data["lambda_rrz"] * data["g_tt"]
-            + 2 * data["lambda_rz"] * data["g_tt_r"]
-            + data["lambda_z"] * data["g_tt_rr"]
-            - data["lambda_rrt"] * data["g_tz"]
-            - 2 * data["lambda_rt"] * data["g_tz_r"]
-            - (1 + data["lambda_t"]) * data["g_tz_rr"]
-            - 2 * num_r * data["sqrt(g)_r"]
-            - num * data["sqrt(g)_rr"]
-        ) / data["sqrt(g)"]
-        den = data["g_tt"] / data["sqrt(g)"]
-        den_r = (data["g_tt_r"] - den * data["sqrt(g)_r"]) / data["sqrt(g)"]
-        den_rr = (
-            data["g_tt_rr"] - 2 * den_r * data["sqrt(g)_r"] - den * data["sqrt(g)_rr"]
-        ) / data["sqrt(g)"]
-        den_avg = surface_averages(transforms["grid"], den)
-        den_avg_r = surface_averages(transforms["grid"], den_r)
-        den_avg_rr = surface_averages(transforms["grid"], den_rr)
-        num_avg_rr = surface_averages(transforms["grid"], num_rr)
         data["iota_rr"] = (
             current_term_rr
-            + num_avg_rr
-            - 2 * data["iota_r"] * den_avg_r
-            - data["iota"] * den_avg_rr
-        ) / den_avg
+            + data["iota_0_num_rr"]
+            - 2 * data["iota_r"] * data["iota_0_den_r"]
+            - data["iota"] * data["iota_0_den_rr"]
+        ) / data["iota_0_den"]
+    return data
+
+
+@register_compute_fun(
+    name="iota_0_num",
+    label="\\iota_{0} numerator",
+    units="m^{-1}",
+    units_long="inverse meters",
+    description="0 current rotational transform numerator",
+    dim=1,
+    params=[],
+    transforms={"grid": []},
+    profiles=[],
+    coordinates="r",
+    data=["lambda_t", "lambda_z", "g_tt", "g_tz", "sqrt(g)"]
+    + ["lambda_rt", "g_tt_rr", "g_tz_r", "g_tz_rr", "sqrt(g)_r", "sqrt(g)_rr"],
+)
+def _iota_0_num(params, transforms, profiles, data, **kwargs):
+    num = (
+        data["lambda_z"] * data["g_tt"] - (1 + data["lambda_t"]) * data["g_tz"]
+    ) / data["sqrt(g)"]
+    data["iota_0_num"] = surface_averages(transforms["grid"], num)
+
+    if transforms["grid"].axis.size:
```

Comment:
Not sure this is the best way to incorporate these, since in most cases we don't need the axis value but the way the dependencies work it will still calculate all the higher order derivatives needed for the axis value.

Reply:
Good point. It seems to avoid unnecessarily computing limits, we should only do so when explicitly specified. I think the most convenient option for the user would be something like:
```python
eq.compute(["iota", "|B|"], axis=True)
```

I plan to add a `limit_params` list to the compute functions.
Then tweak how the dependency list is built to only include `limit_params` 
if `axis` was specified.

Then inside the actual compute function, the condition will be
```python
if axis and transforms.grid.axis.size:
    # replace axis indices with value at limit
```

Reply:
@f0uriest Here's my plan for the API
[limit api.pdf](https://github.com/PlasmaControl/DESC/files/11606378/limit.api.pdf)

Reply:
This looks great! I think its more important to make sure criteria 1&2 are satisfied, ensuring criteria 3 is satisfied seems trickier and could potentially get messy. I'd start with just computing even the limit dependencies over the full grid, and we can see how slow it is. My hunch is it won't be that bad in practice, since most of the optimization grids don't include the axis, so the axis terms would be mainly for post-hoc plotting and analysis where speed isn't as important. It should also simplify the broadcasting etc.

Reply:
we can get a sense of how much extra computation is done by just looking at the ratio of the number of limit dependencies vs regular dependencies for each quantity.